src/stack/marshalling-yard.c

```diff
@@ -0,0 +1,113 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "stack.h"
+
+void append(char* ch, int* capasity, int* lenght, char** str)
+{
+    if (*capasity - 1 <= *lenght)
+    {
+        char* ptr = malloc(sizeof(char) * *capasity * 2);
+        if (ptr == NULL)
+            return;
+
+        *capasity *= 2;
+        strcpy(ptr, *str);
+        free(*str);
+        *str = ptr;
+    }
+    (*str)[*lenght] = *ch;
+    (*lenght)++;
+    (*str)[*lenght] = '\0';
+}
+
+char* reading(int* capasity, int* lenght)
```

Comment:
Функции надо называть глаголами в повелительном наклонении:

```suggestion
char* read(int* capasity, int* lenght)
```